raiden/raiden_service.py

```diff
@@ -240,6 +242,35 @@ def matches(self, token_network_address: TokenNetworkAddress, amount: PaymentAmo
         return token_network_address == self.token_network_address and amount == self.amount
 
 
+class SyncTimeout:
+    """ Helper to determine if the sync should halt or continue.
+
+    The goal of this helper is to stop synching before the block
+    `current_confirmed_head` is pruned, otherwise JSON-RPC requests will start
+    to fail.
+    """
+
+    def __init__(self, current_confirmed_head: BlockNumber, timeout: float) -> None:
+        self.sync_start = datetime.now()
```

Comment:
datetime shouldn't be used to calculate runtimes or timeouts since it's dependent on the system clock.
`time.monotonic()` is much safer.

Reply:
good point, nasty ntp